src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -278,3 +271,42 @@ def open_page_in_browser(url):
         # b. Python 2.x can't sniff out the default browser
         return subprocess.Popen(['open', url])
     return webbrowser.open(url, new=2)  # 2 means: open in a new tab, if possible
+
+
+def _get_platform_info():
+    import platform
+    uname = platform.uname()
+    # python 2, `platform.uname()` returns: tuple(system, node, release, version, machine, processor)
+    platform_name = getattr(uname, 'system', None) or uname[0]
+    release = getattr(uname, 'release', None) or uname[2]
+    return platform_name.lower(), release.lower()
+
+
+def _is_wsl(platform_name, release):
+    platform_name, release = _get_platform_info()
+    return platform_name == 'linux' and release.split('-')[-1] == 'microsoft'
```

Comment:
My only concern is that we're not sure how stable it is.

Reply:
This is proposed by WSL team. If it turns out not stable, CLI will try to launch the browser through the regular flow, so it should be fine.